client/src/pages/Login.js

```diff
@@ -1,16 +1,21 @@
-import React from "react";
+import React, { useState } from "react";
 import styled from "styled-components";
 import { Button, Flex, WhiteSpace } from "antd-mobile";
 import { Link } from "react-router-dom";
+import { makeStyles } from '@material-ui/core/styles';
+import TextField from '@material-ui/core/TextField';
+import { validateEmail } from '../utils/common.js';
+import { useAlert } from 'react-alert';
+import { PASSWORD_MIN_LENGTH } from '../config';
+import { loginWithEmail, signup } from '../actions/authActions';
+import { useDispatch } from "react-redux";
```

Comment:
my initial thoughts about redux were that "we might not need it" and better avoid it to avoid complexity and to increase performance.
For now, I guess it's good to keep it simple and make sure everything works well for the MVP, we could later see if we can just use the React Context API and if we really need a state management solution.